pandas/tests/test_expressions.py

```diff
@@ -431,3 +430,29 @@ def test_bool_ops_column_name_dtype(self, test_input, expected):
         # GH 22383 - .ne fails if columns containing column name 'dtype'
         result = test_input.loc[:, ["a", "dtype"]].ne(test_input.loc[:, ["a", "dtype"]])
         assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "arith", ("add", "sub", "mul", "mod", "truediv", "floordiv")
+    )
+    @pytest.mark.parametrize("axis", (0, 1))
+    def test_frame_series_axis(self, axis, arith):
+        if axis == 1 and arith == "floordiv":
+            pytest.xfail("'floordiv' does not succeed with axis=1 #27636")
+
+        df = self.frame
+        if axis == 1:
+            other = self.frame.iloc[0, :]
+        else:
+            other = self.frame.iloc[:, 0]
+
+        expr._MIN_ELEMENTS = 0
```

Comment:
What does this do?

Reply:
There is an overhead in using numexpr for computations that makes it not worth using for a low number of computations. The values of this defaults to: 10,000. In this test suite we are trying to verify that numexpr is being invoked correctly however the function _can_use_numexpr checks that the objects operated on are sufficient size:

```python
def _can_use_numexpr(op, op_str, a, b, dtype_check):
    """ return a boolean if we WILL be using numexpr """
    if op_str is not None:
        # required min elements (otherwise we are adding overhead)
        if np.prod(a.shape) > _MIN_ELEMENTS:
            # further dtype checks to check compatibility may return True
    return False
```

We can therefore run the test suite exclusively on objects where the number of elements > 10,000 or set the min elements to zero to always try and use numexpr.

This is what is meant by a "large DataFrame". Anything where the size of the objects is large enough to warrant using numexpr fails because of the regression anything that is fairly small will succeed because numexpr evaluation will not even be considered.

Reply:
Hmm is affecting global state, right? I worry about this leaking into other tests...

Can you use the `monkeypatch` fixture, and do this setting with (IIRC) `monkeypatch.setattr`? That way it'll be undone when the test exits.

Reply:
It is but this is handled in the test classes teardown method on line 57 so the global state won't be affected permanently. I can move my test out of the class and handle this with monkeypatch or we can keep the setup/teardown framework already implemented in the TestExpressions class.

Reply:
Thanks, I wasn't familiar with these tests.